### ISSUE TITLE
Querying on a sub-attribute is not supported (unencoded bracket)

### ISSUE THREAD

**Issue by jason-fox** [ScorpioBroker/ScorpioBroker#57]
I create an entity:

```console
curl -L -X POST 'http://localhost:9090/ngsi-ld/v1/entities/' \
-H 'Content-Type: application/ld+json' \
--data-raw '{
    "id": "urn:ngsi-ld:Building:store001",
    "type": "Building",
    "category": {
    	"type": "Property",
        "value": ["commercial"]
    },
    "address": {
        "type": "Property",
        "value": {
            "streetAddress": "Bornholmer Straße 65",
            "addressRegion": "Berlin",
            "addressLocality": "Prenzlauer Berg",
            "postalCode": "10439"
        },
        "verified": {
			"type": "Property",
			"value": true
		}
    },
    "location": {
        "type": "GeoProperty",
        "value": {
             "type": "Point",
             "coordinates": [13.3986, 52.5547]
        }
    },
    "name": {
        "type": "Property",
        "value": "Bösebrücke Einkauf"
    },
    "@context": [
        "https://schema.lab.fiware.org/ld/context"
    ]
}'
```

I make a request using the `q` parameter searching on a sub-attribute - see p36 of the specification

> EXAMPLE 9: address[addressLocality]== "Berlin". The trailing path is [addressLocality] and is
> used to refer to a particular subitem within a Postal Address. 

```console
curl -L -X GET 'http://localhost:1026/ngsi-ld/v1/entities/?type=Building&q=address[addressLocality]==%22Prenzlauer%20Berg%22&options=keyValues' \
-H 'Link: <https://schema.lab.fiware.org/ld/context>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json"' \
-H 'Accept: application/ld+json'
````

Scorpio returns a 400 response with an unformatted error page:

Expected:

```json
[
    {
        "@context": "https://fiware.github.io/data-models/context.jsonld",
        "id": "urn:ngsi-ld:Building:store001",
        "type": "Building",
        "address": {
            "streetAddress": "Bornholmer Straße 65",
            "addressRegion": "Berlin",
            "addressLocality": "Prenzlauer Berg",
            "postalCode": "10439"
        },
        "name": "Bösebrücke Einkauf",
        "category": [
            "commercial"
        ],
        "location": {
            "type": "Point",
            "coordinates": [
                13.3986,
                52.5547
            ]
        }
    }
]
```

**Comment by jason-fox** [comment]
Encoding the square brackets here works fine:

```console
curl -L -X GET 'http://localhost:9090/ngsi-ld/v1/entities/?type=Building&q=address%5BaddressLocality%5D==%22Prenzlauer%20Berg%22&options=keyValues' \
-H 'Link: <https://schema.lab.fiware.org/ld/context>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json"' \
-H 'Accept: application/ld+json'
```

**Comment by jason-fox** [comment]
Related: https://github.com/ScorpioBroker/ScorpioBroker/issues/58

**Comment by ScorpioBroker** [comment]
fixed with issue #58
